tests/network/upgrade/conftest.py

```diff
@@ -72,12 +73,22 @@ def vmb_upgrade_mac_spoof(worker_node1, unprivileged_client, upgrade_linux_macsp
 
 @pytest.fixture(scope="session")
 def running_vma_upgrade_mac_spoof(vma_upgrade_mac_spoof):
-    return running_vm(vm=vma_upgrade_mac_spoof)
+    vma_upgrade_mac_spoof.start(wait=True)
+    vma_upgrade_mac_spoof.vmi.wait_for_condition(
+        condition=VirtualMachineInstance.Condition.Type.AGENT_CONNECTED,
+        status=VirtualMachineInstance.Condition.Status.TRUE,
+    )
```

Comment:
should be:
```suggestion
    vma_upgrade_mac_spoof.wait_for_agent_connected()
```
right?

Reply:
Yes, I have not finished rebasing all yet.
I will take care of the rest soon.

Reply:
Done